libs/datamodel/core/src/ast/reformat/reformatter.rs

```diff
@@ -362,9 +383,65 @@ impl<'a> Reformatter<'a> {
         }
     }
 
+    fn extract_and_sort_directives<'i>(token: &'i Token, is_field_directive: bool) -> Vec<Pair<'i, Rule>> {
+        // get indices of directives and store in separate Vector
+        let mut directive_indices = Vec::new();
+        let mut directives = Vec::new();
+        for (i, pair) in token.clone().into_inner().enumerate() {
+            match pair.as_rule() {
+                Rule::directive => {
+                    directive_indices.push(i);
+                    directives.push(pair);
+                }
+                _ => (),
+            }
+        }
+
+        // sort directives
+        let correct_order = if is_field_directive {
+            vec!["map", "id", "unique", "default", "relation", "updatedAt"]
+        } else {
+            vec!["map", "unique", "index", "id"]
+        };
+        directives.sort_by(|a, b| {
+            let mut sort_index_a = usize::MAX;
+            let mut sort_index_b = usize::MAX;
+            for (i, name) in correct_order.iter().enumerate() {
+                if a.as_str().contains(name) {
+                    sort_index_a = i;
+                    break;
+                }
+            }
```

Comment:
I think you might be able to write that in a shorter way by using the `position` function to find the index: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.position